pyiron_contrib/atomistics/ipi/ipi_core.py

```diff
@@ -136,23 +137,47 @@ def collect_props(self):
         self.custom_output.pressure = np.array([float(i) for i in pressure])
         self.custom_output.energy_tot = self.custom_output.energy_pot + self.custom_output.energy_kin
 
-    def collect_cells(self):
-        f = open(self.working_directory + '/ipi_out.pos_0.xyz')
+    @staticmethod
+    def _collect_traj_helper(filename):
+        f = open(filename)
         lines = f.readlines()
+        starts = []
+        for i, x in enumerate(lines):
+            if x.startswith("#"):
+                starts.append(i)
+        starts.append(len(lines) + 1)
         abc = []
         ABC = []
-        for x in lines:
-            if x.startswith("#"):
-                split_line = x.split()
-                abc.append([float(i) for i in split_line[2:5]])
-                ABC.append([float(i) for i in split_line[5:8]])
+        traj = []
+        start = starts[0]
+        for stop in starts[1:]:
+            temp_list = []
+            snap_lines = lines[start:stop - 1]
+            for i, l in enumerate(snap_lines):
+                split_l = l.split()
+                if i != 0:
+                    temp_list.append([float(j) for j in split_l[1:]])
+                else:
+                    abc.append([float(j) for j in split_l[2:5]])
+                    ABC.append([float(j) for j in split_l[5:8]])
+            start = stop
+            traj.append(temp_list)

```

Comment:
This is a little hard to understand. Could you refactor it out into a well-named method? (It looks like the two previous lines defining `abc` and `ABC` would need to be pulled out at the same time.

I also think that the loop body could be simplified a bit, although do please double check that what I have here is actually doing the same thing, since I don't have data to test on:
```suggestion
        traj = []
        for i, stop in starts[1:]:
            start = starts[i - 1]
            snap_lines = lines[start:stop - 1]
            
            zeroth_line_values = [float(n) for n in snap_lines[0].split()]
            abc.append(zeroth_line_values[2:5])
            ABC.append(zeroth_line_values[5:8])
            # TODO: Comment on what is being grabbed by these slices?      
            
            temp_list = []
            for j, l in enumerate(snap_lines[1:]):
                temp_list.append([float(n) for n in l.split()[1:]])
            traj.append(temp_list)
```

What I like about this is that it gets rid of the `if` clause, making the interior loop super simple s.t. the nested loop is overall quite straightforward.

Reply:
Needed a bit of modification to run properly. Added documentation.